### DISCUSSION TITLE
RLS for a table that depends on another table?

### DISCUSSION THREAD

**Discussion by pierroo** [supabase/supabase#20527] [Questions]
I have a table chat_msg, and a table chat_room

I want to create a RLS for insert on chat_msg, that will make sure I am inserting a msg in a chat_room I belong to
The "belonging" means that either chat_room.invited_uid or chat_room.creator_uid equal my own uid

I tried to do this:

![image](https://github.com/supabase/supabase/assets/17619412/5cc8963f-b2c8-4f93-a6e7-7cb639154c28)

But I get the error "Failed to update database policy: failed to update pg.policies with the given ID: missing FROM-clause entry for table "new"
so I assume the NEW is not accepted like it would be in a trigger

Perhaps it's not the right way to do?
ChatGPT is indeed suggesting me to create a trigger instead to check before insertion, like this:

![image](https://github.com/supabase/supabase/assets/17619412/eddeae9d-7e27-4d94-8b4d-dd90a4c02985)

and have the RLS for insert set to true

What is the correct (and most performant way, since it will be performed on each msg inserted which occurs often) for this use case?

**Comment by GaryAustin1** [comment]
You just refer to the column name by it self or with the main table before the . for RLS.  So I think for your table  chat_msg.id in the where clause.  It is a bit painful when tables have the same column names to keep sorted.
